packages/rlc-common/src/metadata/buildApiExtractorConfig.ts

```diff
@@ -5,16 +5,16 @@ import { Project } from "ts-morph";
 import { RLCModel } from "../interfaces.js";
 
 export function buildApiExtractorConfig(model: RLCModel) {
-  const generateMetadata = Boolean(model.options?.generateMetadata);
-  if (!generateMetadata) {
-    return;
-  }
-  const { generateTest, packageDetails } = model.options || {};
+  let { generateTest, packageDetails, isModularLibrary } = model.options || {};
+  // Take the undefined as true by default
+  generateTest = generateTest === true || generateTest === undefined;
   const project = new Project();
   const config = {
     $schema:
       "https://developer.microsoft.com/json-schemas/api-extractor/v7/api-extractor.schema.json",
-    mainEntryPointFilePath: `./types${generateTest ? "/src" : ""}/index.d.ts`,
+    mainEntryPointFilePath: `./types${
+      generateTest || isModularLibrary ? "/src" : ""
```

Comment:
why isModularLibrary can have impact on this mainEntryPointFilePath ?

Reply:
For modular the generated `index.d.ts` would be in `types/src` folder.

Reply:
even if we don't generate tests and samples ?

Reply:
Yes, this is because in modular tsconfig.json we by default add below option which would always intro a src folder in dist-ems folder.

```
"rootDir": "."
```